zerver/tests/test_users.py

```diff
@@ -908,7 +908,7 @@ def test_create_user_with_multiple_streams(self) -> None:
 
         prereg_user = PreregistrationUser.objects.get(email="fred@zulip.com")
 
-        with self.assert_database_query_count(80):
+        with self.assert_database_query_count(90):
```

Comment:
We should try if we can decrease the queries here.

Reply:
The issue with creating a `PreregistrationUser` is that we cannot pass streams during its creation. Instead, we must add streams later using `prereg_user.streams.set()`. However, this method does not actually set streams with fetched `select_related` fields.
```
  prereg_user = PreregistrationUser(
  email=email,
  referred_by=user_profile,
  invited_as=invite_as,
  realm=realm,
  include_realm_default_subscriptions=include_realm_default_subscriptions,
  notify_referrer_on_join=notify_referrer_on_join,
  )
  prereg_user.save()
  stream_ids = [stream.id for stream in streams]
  prereg_user.streams.set(stream_ids)
```